From 1a33a943c643b43033af936f297898b540361c62 Mon Sep 17 00:00:00 2001
From: Robert Marko <robimarko@gmail.com>
Date: Sat, 1 Jan 2022 19:11:55 +0100
Subject: [PATCH] clk: qcom: ipq8074: disable USB GDSC-s SW_COLLAPSE

Like in IPQ6018 Qualcomm intentionally disables the SW_COLLAPSE on the USB
GDSC-s.

This could potentially be better handled by utilizing the GDSC driver, but
I am not familiar with it nor do I have datasheets.

Signed-off-by: Robert Marko <robimarko@gmail.com>
---
 drivers/clk/qcom/gcc-ipq8074.c | 5 +++++
 1 file changed, 5 insertions(+)

--- a/drivers/clk/qcom/gcc-ipq8074.c
+++ b/drivers/clk/qcom/gcc-ipq8074.c
@@ -4842,6 +4842,11 @@ static int gcc_ipq8074_probe(struct plat
 	/* SW Workaround for UBI32 Huayra PLL */
 	regmap_update_bits(regmap, 0x2501c, BIT(26), BIT(26));
 
+	/* Disable SW_COLLAPSE for USB0 GDSCR */
+	regmap_update_bits(regmap, 0x3e078, BIT(0), 0x0);
+	/* Disable SW_COLLAPSE for USB1 GDSCR */
+	regmap_update_bits(regmap, 0x3f078, BIT(0), 0x0);
+
 	clk_alpha_pll_configure(&ubi32_pll_main, regmap, &ubi32_pll_config);
 	clk_alpha_pll_configure(&nss_crypto_pll_main, regmap,
 				&nss_crypto_pll_config);
